lib/datadog/appsec/component.rb

```diff
@@ -19,12 +20,20 @@ def build_appsec_component(settings)
 
       def initialize(processor: Processor.new)
         @processor = processor
+
+        start_monitor
```

Comment:
What happens when the component gets rebuilt? Are we at risk of having multiple watchers?

This might be good enough for now but to me it highlights that this may not be the correct place as this is instrumentation business and this way we're basically auto-instrumenting.

I'm having some idea, I need to chart it out.

Reply:
I thought about that. 

Currently, the gateway, makes sure the same block is not registered twice.

https://github.com/DataDog/dd-trace-rb/blob/cebb06754a85397fc88f45ff5f172f5364d6d98a/lib/datadog/appsec/instrumentation/gateway.rb#L52